### DISCUSSION TITLE
Re-design deploy-rs integration

### DISCUSSION THREAD

**Discussion by Pacman99** [nrdxp/nixdev#434] [Ideas]
There are two ways deploy-rs is integrated at the moment:
 - You can use `mkDeployNodes` to create a `deploy.nodes` output automatically based on your `nixosConfigurations`
 - If the `deploy` output exists, checks for them will automatically be added.

I think the second integration is great as it is. The first one is a really poorly setup integration. It helps you if you have the simplest use case, but as soon as you want anything more it feels like you have to fight against the function. It assumes the server you want to deploy to can be accessed through the fqdn and that you only want nodes for your `nixosConfigurations` outputs. The function itself is also designed poorly with the way `extraConfig` is implemented and its impractical to add any new features to it, see #81.

Its not that difficult to craft your own `deploy.nodes` output, so I wouldn't be against just removing the `mkDeployNodes` function and just force users to create their own output.

Otherwise either the function should be re-designed to have a cleaner api or perhaps deploy-rs should be fully integrated into digga where we would auto-create the `deploy.nodes` output and possibly even add api arguments to customize how it gets created.

**Comment by blaggacao** [comment]
Hm, it's maybe a good consideration to fully integrate `deploy-rs`. Maybe we can have `digga` organized around `cells` at some point. Then, an optional `deploy-rs` could encapsulate all that functionality. 

In general :+1: for the proposal.
